src/storage/mod.rs

```diff
@@ -5653,4 +5657,167 @@ mod tests {
         let res = rx.recv().unwrap().unwrap();
         assert_eq!(res.min_commit_ts, 1001.into());
     }
+
+    #[test]
+    fn test_overlapped_ts_rollback_before_prewrite() {
```

Comment:
Sorry but I don't understand how this test is related to overlapped ts.

Reply:
It's used to check "rollback happens before commit => should not overlap". I added some comment.